apps/src/music/views/Generate.tsx

```diff
@@ -0,0 +1,195 @@
+import {Button} from '@code-dot-org/component-library/button';
+import React, {useCallback, useState} from 'react';
+
+import Adlib from '@cdo/apps/lab2/views/components/guide/Adlib';
+import Guide from '@cdo/apps/lab2/views/components/guide/Guide';
+import {useAppDispatch, useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import askAi from '../ai/generate/askAi';
+import {generateBlocklyJson} from '../ai/generate/generateBlocklyJson';
+import appConfig from '../appConfig';
+import MusicLibrary from '../player/MusicLibrary';
+import {setCodeToLoad, setAiGenerateState} from '../redux/musicRedux';
+
+import styles from './Generate.module.scss';
+
+const adlibs: {
+  [key: string]: {template: string; options: {[key: string]: string[]}};
+} = {
+  complex: {
+    template:
+      'Please generate a new song.  Around {length} measures in duration is good.  Go for a {mood} vibe.  Sequence it like a {sequence}.',
+    options: {
+      length: ['20', '30', '40'],
+      mood: ['happy', 'sad', 'energetic', 'calm', 'upbeat', 'chill'],
+      sequence: ['verse-chorus-verse', 'A-B-A', 'A-B-A-C-A', 'A-A-B-A'],
+    },
+  },
+  length: {
+    template:
+      'Please generate a new song.  Around {length} measures in duration is good.',
+    options: {
+      length: ['20', '30', '40'],
+    },
+  },
+  sounds: {
+    template: 'Please make a song with {sounds} sounds in order.',
+    options: {
+      sounds: ['2', '3', '4', '5'],
+    },
+  },
+  layers: {
+    template:
+      'Please make a song that layers {type1} and {type2} together. Around 10 measures is good.',
+    options: {
+      type1: ['bass', 'leads', 'beats', 'vocals'],
+      type2: ['bass', 'leads', 'beats', 'vocals'],
+    },
+  },
+};
+
+interface GenerateProps {}
+
+const Generate: React.FunctionComponent<GenerateProps> = () => {
```

Comment:
nit: can remove the `GenerateProps` interface if we don't have any props (the type can just be `React.FunctionComponent` with no parameter)

Reply:
Done.